### PR TITLE
fix(predicted-latency-scorer): prevent nil pointer panic for non-completions API types

### LINKED ISSUE #2394: predicted-latency-scorer crashes on /v1/chat/completions requests
**What happened**:

When sending a `/v1/chat/completions` request with the `predicted-latency-scorer` plugin enabled, the EPP crashes with a nil pointer dereference panic:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1e51b54]

goroutine 734 [running]:
sigs.k8s.io/gateway-api-inference-extension/pkg/epp/framework/plugins/scheduling/scorer/predictedlatency.(*PredictedLatency).generatePredictions(...)
prediction.go:66
```
**What you expected to happen**:

The predicted-latency-scorer plugin should handle both `/v1/completions` and `/v1/chat/completions` requests without crashing.

**How to reproduce it (as minimally and precisely as possible)**:

1. Enable the `predicted-latency-scorer` in the EPP configuration
2. Send a chat completions request
3. EPP crashes immediately with nil pointer dereference (/v1/completions requests work correctly)

**Anything else we need to know?**:
Add a helper method to `LLMRequestBody` that extracts prompt text from either `Completions.Prompt` or `ChatCompletions.Messages`, so latency prediction works correctly for both API types.

**Environment**:
Kubernetes version: v1.32.5
Inference extension version: https://github.com/kubernetes-sigs/gateway-api-inference-extension/commit/74b7a04b76526fd8e2e2d5897e978aa96887a44a

### PR DESCRIPTION
**Pull request by noalimoy**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

The `predicted-latency-scorer` plugin crashes with a nil pointer dereference (`SIGSEGV`) when processing `/v1/chat/completions` requests.
The root cause is that `generatePredictions`, `recordTTFTTrainingData`, and `processTokenForLatencyPrediction` all access `Body.Completions.Prompt`
directly — which is `nil` for any non-`/v1/completions` API type.

This PR:
- Adds `LLMRequestBody.PromptText()` — a safe helper that extracts prompt text from whichever API type is present (`Completions`, `ChatCompletions`, `Responses`, `Conversations`), following the same switch-based pattern as
  the existing `CacheSalt()` method.
- Replaces all 4 direct `Body.Completions.Prompt` accesses in the `predictedlatency` scorer with `Body.PromptText()` calls.
- Adds comprehensive unit tests:
  - `TestLLMRequestBody_PromptText` — 9 subtests covering all API types and edge cases.
  - `TestPredictedLatency_Score/Chat_completions_request_does_not_panic` — scorer-level regression test.
  - `TestBulkPredictWithMetrics_ChatCompletionsPrompt` — helper-level regression test.

**Which issue(s) this PR fixes**:
Fixes #2394

**Does this PR introduce a user-facing change?**:
Fix nil pointer panic in` predicted-latency-scorer` when processing /v1/chat/completions, /v1/responses, and /v1/conversations requests.

### CODE REVIEW

**Comment by noalimoy** [comment]
Hey @bongwoobak — would be happy to get your review on the PR!

File: pkg/epp/framework/plugins/scheduling/scorer/predictedlatency/prediction.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ func (s *PredictedLatency) generatePredictions(ctx context.Context, request *sch
 
 		metricsStates[i] = endpoint.GetMetrics()
 		targetEndpointsMetadatas[i] = endpoint.GetMetadata()
-		prompts[i] = request.Body.Completions.Prompt
+		prompts[i] = request.Body.PromptText()
 		generatedTokenCounts[i] = 1
 		prefixCacheScores[i] = prefixCacheScore
 	}
```
Comment (bongwoobak) [inline pkg/epp/framework/plugins/scheduling/scorer/predictedlatency/prediction.go:69]: Now that `PromptText()` does actual work (string concatenation, JSON marshaling) instead of a plain field access, it might be worth hoisting the call out of the loop since the result is the same for every candidate endpoint.

like

```suggestion
	promptText := request.Body.PromptText()
	for i, endpoint := range candidateEndpoints {
		logger.V(logutil.TRACE).Info("Candidate pod for scheduling", "endpoint", endpoint.GetMetadata().String(), "metrics", endpoint.GetMetrics().String())

		// Get prefix cache score for the pod
		prefixCacheScore := predictedLatencyCtx.prefixCacheScoresForEndpoints[endpoint.GetMetadata().NamespacedName.Name]

		logger.V(logutil.DEBUG).Info("Prefix cache score for pod", "pod", endpoint.GetMetadata().String(), "prefixCacheScore", prefixCacheScore)

		metricsStates[i] = endpoint.GetMetrics()
		targetEndpointsMetadatas[i] = endpoint.GetMetadata()
		prompts[i] = promptText
		generatedTokenCounts[i] = 1
		prefixCacheScores[i] = prefixCacheScore
	}
```

Comment (kaushikmitr) [reply pkg/epp/framework/plugins/scheduling/scorer/predictedlatency/prediction.go:69]: Thanks for fixing this. agreed, json.marshall can be expensive which this method sometimes uses. I would suggest while creating the predictionrequestcxt, update the prompt field with this and use it subsequently. Actually i was working on something similar but did not get a chance yet to create PR. https://github.com/tomatillo-and-multiverse/gateway-api-inference-extension-merged/blob/6b3375e759db82e9c7ec88c069ebf51f8c07d19a/pkg/epp/framework/plugins/scheduling/scorer/predictedlatency/requestcontrol_hooks.go#L87

Comment (noalimoy) [reply pkg/epp/framework/plugins/scheduling/scorer/predictedlatency/prediction.go:69]: Thanks for the suggestion @kaushikmitr — I agree that caching the prompt text in `predictedLatencyCtx` would avoid redundant `PromptText()` calls.

Implemented — I added `PromptText()` as a method on `LLMRequestBody` in `types.go` (alongside `CacheSalt()`) rather than a package-private helper, so any plugin can reuse it without importing the scorer package.

**Comment by kaushikmitr** [comment]
@noalimoy can you please fix the lint error
